NorthstarDLL/httprequesthandler.cpp

```diff
@@ -0,0 +1,504 @@
+
+	// This handle will be returned to Squirrel so it can wait for the response and assign a callback for it.
+	int handle = ++m_iLastRequestHandle;
+
+	std::thread requestThread(
+		[this, handle, requestParameters, bAllowLocalHttp]()
+		{
+			std::string hostname, resolvedAddress, resolvedPort;
+
+			if (!bAllowLocalHttp)
+			{
+				if (!IsHttpDestinationHostAllowed(requestParameters.baseUrl, hostname, resolvedAddress, resolvedPort))
+				{
+					spdlog::warn(
+						"HttpRequestHandler::MakeHttpRequest attempted to make a request to a private network. This is only allowed when "
+						"running the game with -allowlocalhttp.");
+
+					g_pSquirrel<context>->schedule_call("NSHandleFailedHttpRequest", handle, (int)0, "Cannot make HTTP requests to private network hosts without -allowlocalhttp. Check your console for more information.");			
+					return;
+				}
+			}
+
+			CURL* curl = curl_easy_init();
+			if (!curl)
+			{
+				spdlog::error("HttpRequestHandler::MakeHttpRequest failed to init libcurl for request.");
+				g_pSquirrel<context>->schedule_call(
+					"NSHandleFailedHttpRequest", handle, static_cast<int>(CURLE_FAILED_INIT), curl_easy_strerror(CURLE_FAILED_INIT));				
+				return;
+			}
+
+			// HEAD has no body.
+			if (requestParameters.method == HttpRequestMethod::HRM_HEAD)
+			{
+				curl_easy_setopt(curl, CURLOPT_NOBODY, 1L);
+			}
+				
+			curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, HttpRequestMethod::ToString(requestParameters.method).c_str());
+
+			// Only resolve to IPv4 if we don't allow private network requests.
+			curl_slist* host = nullptr;
+			if (!bAllowLocalHttp)
+			{
+				curl_easy_setopt(curl, CURLOPT_IPRESOLVE, CURL_IPRESOLVE_V4);
+				host = curl_slist_append(host, fmt::format("{}:{}:{}", hostname, resolvedPort, resolvedAddress).c_str());
+				curl_easy_setopt(curl, CURLOPT_RESOLVE, host);
+			}
+
+			// Ensure we only allow HTTP or HTTPS.
+			curl_easy_setopt(curl, CURLOPT_PROTOCOLS, CURLPROTO_HTTP | CURLPROTO_HTTPS);
+
+			// Allow redirects
+			curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1L);
+			curl_easy_setopt(curl, CURLOPT_MAXREDIRS, 3L);
+
+			// Check if the url already contains a query.
+			// If so, we'll know to append with & instead of start with ?
+			std::string queryUrl = requestParameters.baseUrl;
+			bool bUrlContainsQuery = false;
+
+			// If this fails, just ignore the parsing and trust what the user wants to query.
+			// Probably will fail but handling it here would be annoying.
+			CURLU* curlUrl = curl_url();
+			if (curlUrl)
+			{
+				if (curl_url_set(curlUrl, CURLUPART_URL, queryUrl.c_str(), CURLU_DEFAULT_SCHEME) == CURLUE_OK)
+				{
+					char* currentQuery;
+					if (curl_url_get(curlUrl, CURLUPART_QUERY, &currentQuery, 0) == CURLUE_OK)
+					{
+						if (currentQuery && std::strlen(currentQuery) != 0)
+						{
+							bUrlContainsQuery = true;
+						}
+					}
+
+					curl_free(currentQuery);
+				}
+
+				curl_url_cleanup(curlUrl);
+			}
+
+			curlUrl = nullptr;
```

Comment:
Variable "curlUrl" is deleted and past deletion, no longer used. Assignment to nullptr is unnecessary.

Reply:
Resolved on new branch, will re-make PR.